app/javascript/components/calendar/agenda-multi-agent.js

```diff
@@ -54,6 +69,48 @@ class AgendaMultiAgent {
     });
     window.location = `/admin/organisations/${this.data.organisationId}/rdv_wizard_step/new?${urlSearchParams.toString()}`;
   }
+ 
+  datesSet = (info) => {
+    localStorage.setItem("chosenCalendarView", info.view.type);
+    localStorage.setItem("chosenCalendarDay", info.startStr?.split("T")[0]);
+    this.refreshColumnsVisualGrouping();
+  }
+
+  toggleGrouping = () => {
+    localStorage.setItem("groupByDate", this.getGroupByDate() ? "false" : "true");
+    this.fullCalendarInstance.setOption("datesAboveResources", this.getGroupByDate());
+    this.fullCalendarInstance.setOption("customButtons", this.customButtons());
+    this.refreshColumnsVisualGrouping();
+  }
+
+  refreshColumnsVisualGrouping = () => {
+    const allColumns = document.querySelectorAll(".fc-timegrid-col.fc-day");
+    const WHITE = "#FFF";
+    const GREY = "rgb(227, 234, 239, 0.5)";
+
+    if (this.fullCalendarInstance.view.type !== "resourceTimeGridWeek") {
+      return allColumns.forEach(column => column.style.backgroundColor = WHITE); // Reset to white
+    }
+
+    const groupingCriteria = column => this.getGroupByDate() ? column.dataset.date : column.dataset.resourceId;
```

Comment:
Je trouve ça un tout petit peu compliqué à lire dans cet état. Est-ce que le linter autorise des parenthèses dans ce style ?
```suggestion
    const groupingCriteria = (column) => (this.getGroupByDate() ? column.dataset.date : column.dataset.resourceId);
```

Reply:
:heavy_check_mark: